File: 41-first-missing-positive/solution.java
// 253 ms | 251.2 MB
class Solution {
    public int firstMissingPositive(int[] nums) {
        int[]freq=new int[10000002];
        for(int i=0;i<nums.length;i++){
            if(nums[i]>0 && nums[i]<=nums.length){
                freq[nums[i]]++;
            }
        }
        for(int i=1;i<=nums.length;i++){
            if(freq[i]==0)return i;
        }
        return nums.length +1;
    }
}

File: 41-first-missing-positive/solution.cpp
// 0 ms | 54.9 MB
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {

        for(int i = 0; i < nums.size(); i++){
            int element = nums[i];

            if(element >= 1 && element <= nums.size()){
                int current_element = element - 1;

                if(element != nums[current_element]){
                    swap(nums[i], nums[current_element]);
                    i--;
                }
            }
        }

        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != i + 1){
                return i + 1;
            }
        }

        return nums.size() + 1;
    }
};